synapse/tests/test_model_infotech.py

```diff
@@ -234,3 +236,334 @@ def test_model_infotech_hostexec(self):
+                valu, subs = core.getTypeNorm('it:semver', s)
+                self.eq(valu, ev)
+                self.eq(subs, es)
+
+            # norm ints
+            data = (
+                (0, {'major': 0, 'minor': 0, 'patch': 0}),
+                (1, {'major': 0, 'minor': 0, 'patch': 1}),
+                (2, {'major': 0, 'minor': 0, 'patch': 2}),
+                (0xFFFFFFFF, {'major': 0, 'minor': 0, 'patch': 0xFFFFFFFF}),
+                (0xFFFFFFFF + 1, {'major': 0, 'minor': 1, 'patch': 0}),
+                (0xdeadb33f1337733101020304, {'major': 0xdeadb33f, 'minor': 0x13377331, 'patch': 0x01020304}),
+                (0xFFFFFFFFFFFFFFFFFFFFFFFF, {'major': 0xFFFFFFFF, 'minor': 0xFFFFFFFF, 'patch': 0xFFFFFFFF})
+            )
+            for intval, e in data:
+                valu, subs = core.getTypeNorm('it:semver', intval)
+                self.eq(valu, intval)
+                self.eq(subs, e)
+
+            # Bad strings / ints
+            badsemvers = (
+                # Invalid strings
+                '1',
+                '1.2',
+                '2.0A1',
+                '0.18rc2',
+                '0.0.00001',
+                '2016-03-01',
+                'v2.4.0.0-1',
+                '1.3a2.dev12',
+                'OpenSSL_1_0_2l',
+                '1.2.windows-RC1',
+                'v2.4.1.0-0.3.rc1',
+                # invalid ints
+                -1,
+                0xFFFFFFFFFFFFFFFFFFFFFFFF + 1,
+                # Invalid build and prerelease values
+                '1.2.3-alpha.foo..+001',
+                '1.2.3-alpha.foo.001+001',
+                '1.2.3-alpha+001.blahblahblah...',
+                '1.2.3-alpha+001.blahblahblah.*iggy',
+                # Just bad input
+                (),
+                '   ',
+                ' alpha ',
+
+            )
+            for valu in badsemvers:
+                self.raises(BadTypeValu, core.getTypeNorm, 'it:semver', valu)
+
+            reprs = (
+                (0, '0.0.0'),
+                (1, '0.0.1'),
+                (0x000000010000000200000003, '1.2.3'),
+            )
+            for v, e in reprs:
+                self.eq(core.getTypeRepr('it:semver', v), e)
+
+    def test_model_infotech_brutecast(self):
+        with self.getRamCore() as core:
+            # Try some integers
+            valu = core.getTypeCast('it:version:brute', s_version.packVersion(1))
+            self.eq(valu, 0x000000010000000000000000)
+
+            valu = core.getTypeCast('it:version:brute', s_version.packVersion(1, 2, 255))
+            self.eq(valu, 0x0000000100000002000000FF)
+
+            valu = core.getTypeCast('it:version:brute', '1')
+            self.eq(valu, 0x000000010000000000000000)
+
+            # A valid semver
+            valu = core.getTypeCast('it:version:brute', '1.2.3-B5CD5743F')
+            self.eq(valu, 0x000000010000000200000003)
+
+            valu = core.getTypeCast('it:version:brute', '1.2.3-beta+exp.sha.5114f85')
```

Comment:
Does this feel like we should change getTypeCast should return valu, subs ?

Reply:
To be honest it felt really weird to have getTypeCast not return a valu, subs tuple.  If we change it  to do so, we'll need to trickle that down into the ingest use of cast to return only the valu and drop the subs (or figure out a way to represent subs in the ingest scope somehow)

Reply:
backlogged for us